aiohttp/helpers.py

```diff
@@ -748,7 +748,11 @@ def _parse_content_type(self, raw: str) -> None:
             self._content_type = "application/octet-stream"
             self._content_dict = {}
         else:
-            self._content_type, self._content_dict = cgi.parse_header(raw)
+            msg = HeaderParser().parsestr("Content-Type: " + raw)
+            self._content_type = msg.get_content_type()
```

Comment:
One difference compared to cgi is that if the actual type is invalid (for example `foo`), this returns `text/plain`.

Reply:
Documentation here suggest to use `Message`: https://peps.python.org/pep-0594/#cgi
Though the result looks much the same.

Reply:
Yeah, I know.

But to me the _example_ (it's labeled as such, not a suggestion) in the PEP isn't that great -- to parse a HTTP header one would create an e-mail Message, set a header to it and then grab the header's parameters back... quite ugly in my opinion. The approach I took here to me reads much better: parse a header using a header parser.

Another reason using a Message doesn't sound that great to me in new code is that it is labeled as a compat/legacy thing:
- https://docs.python.org/3/library/email.compat32-message.html#compat32-message
- https://docs.python.org/3/library/email.message.html#id2